specs/core/0_fork-choice.md

```diff
@@ -63,17 +63,30 @@ class Target(object):
     root: Hash
 ```
 
+#### `RootSlot`
+
+```python
+@dataclass
+class RootSlot(object):
+    slot: Slot
+    root: Hash
+```
+
+
+
 #### `Store`
 
 ```python
 @dataclass
 class Store(object):
     blocks: Dict[Hash, BeaconBlock] = field(default_factory=dict)
-    states: Dict[Hash, BeaconState] = field(default_factory=dict)
+    states: Dict[RootSlot, BeaconState] = field(default_factory=dict)
     time: int = 0
     latest_targets: Dict[ValidatorIndex, Target] = field(default_factory=dict)
     justified_root: Hash = ZERO_HASH
+    justified_epoch: Epoch = 0
```

Comment:
`justified_root` and `justified_epoch` can be combined into `justified_checkpoint`. Same for `finalized_root` and `finalized_epoch`.